homeassistant/components/roomba/combo.py

```diff
@@ -0,0 +1,159 @@
+"""Class for Combo devices."""
+import logging
+
+from homeassistant.components.vacuum import VacuumEntityFeature
+
+from .irobot_base import SUPPORT_IROBOT, IRobotVacuum
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BIN_DETECTED = "detected_bin"
+ATTR_PAD_WETNESS = "spray_amount"
+ATTR_NAV_MODE = "navigation_mode"
+ATTR_VACC_POWER = "vacuuming_power"
+
+LOW_POWER = "Low"
+HIGH_POWER = "High"
+
+NAV_REACTIVE = "Reactive"
+NAV_STRAIGHT_LINE = "Straight"
+
+COMBO_LIQUID_ECO = "Eco"
+COMBO_LIQUID_STANDARD = "Standard"
+COMBO_LIQUID_ULTRA = "Ultra"
+
+# NAV_REACTIVE = 0; NAV_STRAIGHT_LIVE = 1
+COMBO_NAVIGATION_MODES = [NAV_REACTIVE, NAV_STRAIGHT_LINE]
+COMBO_VACUUM_POWER = [LOW_POWER, HIGH_POWER]
+COMBO_LIQUID_AMOUNT = [COMBO_LIQUID_ECO, COMBO_LIQUID_STANDARD, COMBO_LIQUID_ULTRA]
+
+SUPPORT_COMBO = SUPPORT_IROBOT | VacuumEntityFeature.FAN_SPEED
+
+
+class RoombaCombo(IRobotVacuum):
+    """Combo."""
+
+    def __init__(self, roomba, blid):
+        """Initialize the Roomba handler."""
+        super().__init__(roomba, blid)
+
+        # Initialize fan speed list
+        speed_list = []
+        for behavior in COMBO_VACUUM_POWER:
+            for spray in COMBO_LIQUID_AMOUNT:
+                for mode in COMBO_NAVIGATION_MODES:
+                    speed_list.append(f"{behavior} - {spray} - {mode}")
```

Comment:
It's a bit weird to create a long fan speed list with permutations of the features.

I would suggest to create a new discussion here, suggesting to add additional features to the vacuum platform: https://github.com/home-assistant/architecture/discussions?discussions_q=is%3Aopen+vacuum